src/openforms/js/components/admin/form_design/registrations/objectsapi/fields/DocumentTypes.js

```diff
@@ -1,28 +1,194 @@
+import {useField, useFormikContext} from 'formik';
+import PropTypes from 'prop-types';
 import {FormattedMessage} from 'react-intl';
+import {useUpdateEffect} from 'react-use';
+import useAsync from 'react-use/esm/useAsync';
 
+import Field from 'components/admin/forms/Field';
 import Fieldset from 'components/admin/forms/Fieldset';
+import FormRow from 'components/admin/forms/FormRow';
+import ReactSelect from 'components/admin/forms/ReactSelect';
+import {get} from 'utils/fetch';
 
-import CatalogueSelect from './CatalogueSelect';
+import CatalogueSelect, {extractValue as getCatalogueOption} from './CatalogueSelect';
 
-export const DocumentTypesFieldet = () => (
-  <Fieldset
-    title={
-      <FormattedMessage
-        description="Objects registration: document types"
-        defaultMessage="Document types"
-      />
-    }
-    collapsible
-    fieldNames={['catalogue']}
-  >
-    <div className="description">
-      <FormattedMessage
-        description="New document types informative message"
-        defaultMessage={`The legacy document types configuration will be ignored
+// Data fetching
+
+const CATALOGUES_ENDPOINT = '/api/v2/registration/plugins/objects-api/catalogues';
+const IOT_ENDPOINT = '/api/v2/registration/plugins/objects-api/informatieobjecttypen';
+
+const getCatalogues = async apiGroupID => {
+  const response = await get(CATALOGUES_ENDPOINT, {objects_api_group: apiGroupID});
+  if (!response.ok) {
+    throw new Error('Loading available catalogues failed');
+  }
+  const catalogues = response.data;
+
+  const _optionsByRSIN = {};
+  for (const catalogue of catalogues) {
+    const {rsin} = catalogue;
+    if (!_optionsByRSIN[rsin]) _optionsByRSIN[rsin] = [];
+    _optionsByRSIN[rsin].push(catalogue);
+  }
+
+  const groups = Object.entries(_optionsByRSIN)
+    .map(([rsin, options]) => ({
+      label: rsin,
+      options: options.sort((a, b) => a.label.localeCompare(b.label)),
+    }))
+    .sort((a, b) => a.label.localeCompare(b.label));
+
+  return groups;
+};
+
+const getDocumentTypes = async (apiGroupID, catalogueUrl) => {
+  const response = await get(IOT_ENDPOINT, {
+    objects_api_group: apiGroupID,
+    catalogus_url: catalogueUrl,
+  });
+  if (!response.ok) {
+    throw new Error('Loading available document types failed');
+  }
+  const documentTypes = response.data.sort((a, b) => a.omschrijving.localeCompare(b.omschrijving));
+  return documentTypes.map(({omschrijving}) => ({
+    value: omschrijving,
+    label: omschrijving,
+  }));
+};
+
+// Components
+
+const DocumentType = ({name, label, loading, options, isDisabled, helpText}) => {
+  const [, , fieldHelpers] = useField(name);
+  const {setValue} = fieldHelpers;
+  return (
+    <FormRow>
+      <Field name={name} label={label} helpText={helpText} noManageChildProps>
+        <ReactSelect
+          name={name}
+          options={options}
+          isLoading={loading}
+          isDisabled={isDisabled}
+          onChange={selectedOption => {
```

Comment:
this is because the backend PR needs to be merged first, if I remember correctly

Reply:
it was not because of the backend, fixed now :) some sychronization issue with catalogues URLs being resolved too late and form state being updated/clearing values.